src/library.py

```diff
@@ -910,82 +910,261 @@ def verify(self):
+
+        vacancy_count = 0
+        for sch in occ_df:
+            if sch <= tol:
+                vacancy_count += 1
+            else:
+                vacancy_count = 0
+            if vacancy_count == no_of_consecutive_time_steps:
+                return True
 
+        return False
 
-class GuestRoomControlVent(CheckLibBase):
+    def verify(self):
+        time_step = (
+            self.df.index[1] - self.df.index[0]
+        ).seconds / 60  # time step in min
+        T_z_hea_set_occ = self.df["T_z_hea_set_occ"][0]
+        T_z_coo_set_occ = self.df["T_z_coo_set_occ"][0]
+        tol_occ = self.df["tol_occ"][0]
+        tol_temp = self.df["tol_temp"][0]
+        year_info = 2000
+        result_repo = []
+        for idx, day in self.df.groupby(self.df.index.date):
+            if (
+                day.index.month[0] == 2 and day.index.day[0] == 29
+            ):  # skip leap year, although E+ doesn't have leap year the date for loop assumes so because 24:00 time step so, it's intentionally skipped here
+                pass
+            elif (
+                year_info != day.index.year[0]
+            ):  # remove the Jan 1st of next year reason: the pandas date for loop iterates one more loop is hour is 24:00:00
+                pass
+            else:
+                self.determine_30_min_vacancy(day["O_sch"], time_step)
+                if (day["O_sch"] > tol_occ).any():  # confirmed this room is rented out
+                    if (
+                        day["T_z_hea_set"] < T_z_hea_set_occ - 2.22 + tol_temp
+                    ).any() or (
+                        day["T_z_coo_set"] > T_z_coo_set_occ + 2.22 - tol_temp
+                    ).any():
+                        result_repo.append(
+                            True
+                        )  # confirm the HVAC setpoint control resets when guest room reset when occupants leave the room
+                    else:
+                        result_repo.append(
+                            False
+                        )  # reset does not meet the standard or no reset was observed.
+
+                else:  # room is NOT rented out
+                    if self.determine_30_min_vacancy(day["O_sch"], time_step):
+                        if (day["T_z_hea_set"] < 15.6 + tol_temp).any() and (
+                            day["T_z_coo_set"] > 26.7 - tol_temp
+                        ).any():
+                            result_repo.append(
+                                True
+                            )  # confirmed zone temperature setpoint reset during the unrented period
+                        else:
+                            result_repo.append(
+                                False
+                            )  # zone temperature setpoint was not reset correctly
+                    else:
+                        result_repo.append(True)
+                year_info = day.index.year[0]
+
+        dti = pd.date_range("2020-01-01", periods=365, freq="D")
```

Comment:
it looks like this would fail if the simulation data does not cover 365 days. I suggest you get index from simulation data directly.